### ISSUE TITLE
chrome problem

### ISSUE THREAD

**Issue by jameslarking** [MarcDiethelm/jQuery-Shorten#4]
Hi Marc,

Any idea why the following very simple example does not work in chrome (but does in FF) and whether this is a known issue or a misunderstanding of what I think the plugin should do.

http://dl.dropbox.com/u/7969407/shortenNotWorking.html

**Comment by MarcDiethelm** [comment]
After some initial testing all I can say is: This is really, really weird! In Chrome the "native" CSS property text-overflow:ellipsis is used. But in this basic use case it doesn't behave as expected at all.

1) It seems to do nothing on an inline element (like "A" or "SPAN").
2) On block level elements it only seems to work if:
white-space: nowrap;
overflow: hidden;
are set.

May well be a bug in webkit... Will do some more testing as soon as possible.

**Comment by MarcDiethelm** [comment]
Fixed.

Actually the CSS3 spec requires the element to be a block for text-overflow. And Chrome additionally requires white-space:nowrap and overflow:hidden. After much testing of many different scenarios I came up with the following solution:

Let Shorten set all selected elements to display:block and white-space:nowrap. When using text-overflow it's also setting the element to overflow:hidden.

So far I have not encountered any regressions only a big improvement in supported use cases. I will add my new tests to the repository soon.

Thanks for the bug report!
